modules/core/nomad_clients.tf

```diff
@@ -36,11 +36,25 @@ module "nomad_clients" {
 # ---------------------------------------------------------------------------------------------------------------------
 
 module "consul_iam_policies_clients" {
-  source = "github.com/hashicorp/terraform-aws-consul//modules/consul-iam-policies?ref=v0.3.1"
+  source = "github.com/hashicorp/terraform-aws-consul//modules/consul-iam-policies?ref=v0.3.5"
 
   iam_role_id = "${module.nomad_clients.iam_role_id}"
 }
 
+# ---------------------------------------------------------------------------------------------------------------------
+# PERMIT CONSUL SPECIFIC TRAFFIC IN NOMAD CLUSTER
+# To allow our Nomad clients consul agents to communicate with other consul agents and participate in the LAN gossip,
+# we open up the consul specific protocols and ports for consul traffic
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "nomad_client_consul_gossip" {
+  source = "github.com/hashicorp/terraform-aws-consul.git//modules/consul-client-security-group-rules?ref=v0.3.5"
+
+  security_group_id                  = "${module.nomad_clients.security_group_id}"
+  allowed_inbound_cidr_blocks        = "${concat(list(module.vpc.vpc_cidr_block), var.nomad_clients_allowed_inbound_cidr_blocks)}"
+  allowed_inbound_security_group_ids = ["${module.consul_servers.security_group_id}"]
+}
```

Comment:
Curious qn: Why wasn't this module required before?

Side note nitpick: You might want to change `github.com/hashicorp/terraform-aws-consul.git` to `github.com/hashicorp/terraform-aws-consul` for consistency. `nomad_server_consul_gossip` has the same issue.

Reply:
Okay